ethcore/evm/src/interpreter/mod.rs

```diff
@@ -525,6 +558,25 @@ impl<Cost: CostType> Interpreter<Cost> {
 			instructions::JUMPDEST => {
 				// ignore
 			},
+			instructions::JUMPSUB => {
+				if self.return_stack.len() >= self.return_stack_limit - 1 {
+					return Err(vm::Error::OutOfReturnStack)
+				}
+				let jump = self.stack.pop_back();
+				if jump >= U256::from(usize::max_value()) {
+					return Err(vm::Error::InvalidSub)
```

Comment:
why do we need a separate error type for this?

Reply:
We can use errors from `JUMPDEST` of course, but I think using separate error types is just more descriptive. It does not hurt anyway!

Reply:
Sorry, I meant why not using `BadSubDestination` as we do return `BadJumpDestination` for `verify_jump` in such a case?

Reply:
Hmm yeah you're right!